lib/hanami/validations/form.rb

```diff
@@ -14,7 +14,18 @@ module Validations
     #
     #   class Signup
     #     include Hanami::Validations::Form
+    #
+    #     validations do
+    #       required(:name).filled(:foo?)
+    #       optional(:location).maybe(:str?)
+    #     end
     #   end
+    #
+    #   result = Signup.new('location' => 'Rome').validate
+    #   result.success? # => false
+    #
+    #   result = Signup.new('name' => 'Luca').validate
```

Comment:
I think using symbols as keys works too (needs checking). If so, it might be good to put an example with each key format.